website/data-storage/src/programs/programs.service.ts

```diff
@@ -84,7 +85,7 @@ export class ProgramsService {
       return program;
     } catch (error) {
       logger.error(error);
-      return null;
+      throw new ProgramNotFound(error.message);
```

Comment:
And one more thing
It's necessary to handle this exception in the places where this method is called. (setStatus, isInDb, addProgramInfo methods and a few more)

Reply:
@xhjkl What about this?